fdbserver/LogSystemPeekCursor.actor.cpp

```diff
@@ -895,8 +934,10 @@ ILogSystem::SetPeekCursor::SetPeekCursor(std::vector<Reference<LogSet>> const& l
 	for (int i = 0; i < logSets.size(); i++) {
 		for (int j = 0; j < logSets[i]->logServers.size(); j++) {
 			bool returnEmptyIfStopped =
-			    ((SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST && end != std::numeric_limits<Version>::max())
-			         ? (bestSet < 0 || bestServer < 0 || (i == bestSet && j == bestServer))
+			    ((SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST && bestSet >= 0 &&
```

Comment:
The previous iteration of this logic set `returnEmptyIfStopped` to true if `bestSet < 0`. Now you are requiring `bestServer >=0`, so the logic has changed to never be true is `bestSet < 0`. Is this what you want? You could remove `bestSet >= 0` check and have the helper make the decision.